src/main/java/org/folio/marc4ld/service/ld2marc/processing/combine/impl/DataFieldCombinerFactoryImpl.java

```diff
@@ -1,23 +1,36 @@
 package org.folio.marc4ld.service.ld2marc.processing.combine.impl;
 
+import static org.folio.ld.dictionary.ResourceTypeDictionary.DISSERTATION;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.INSTANCE;
+
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
 import org.folio.marc4ld.service.ld2marc.processing.combine.DataFieldCombiner;
 import org.folio.marc4ld.service.ld2marc.processing.combine.DataFieldCombinerFactory;
 import org.marc4j.marc.DataField;
-import org.springframework.stereotype.Component;
 
-@Component
 public class DataFieldCombinerFactoryImpl implements DataFieldCombinerFactory {
 
+  private final Map<Predicate<DataFieldCombiner.Context>, DataFieldCombiner> combinerMap;
+
+  public DataFieldCombinerFactoryImpl() {
+    combinerMap = Map.of(
+      c -> c.marcTag().equals("245") && c.resourceTypes().contains(INSTANCE), new TitleCombiner(),
```

Comment:
245 & 010 has to be combined after generating marc tags for the entire INSTANCE.
502 has to be combined for each DISSERTATION resource.